templates/trade-confirmation/pdf_trade_confirmation.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"log"
+	"os"
+	"text/template"
+
+	"github.com/unidoc/unipdf/v3/common"
+	"github.com/unidoc/unipdf/v3/common/license"
+	"github.com/unidoc/unipdf/v3/core"
+	"github.com/unidoc/unipdf/v3/creator"
+)
+
+func init() {
+	// Make sure to load your metered License API key prior to using the library.
+	// If you need a key, you can sign up and create a free one at https://cloud.unidoc.io.
+	err := license.SetMeteredKey(os.Getenv(`UNIDOC_LICENSE_API_KEY`))
+	if err != nil {
+		panic(err)
+	}
+
+	common.SetLogger(common.NewConsoleLogger(common.LogLevelDebug))
+}
+
+func main() {
+	c := creator.New()
+	c.SetPageMargins(50, 50, 25, 25)
+
+	// Read main content template.
+	mainTpl, err := readTemplate("templates/main.tpl")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Draw main content teplate.
+	tplOpts := &creator.TemplateOptions{
+		HelperFuncMap: template.FuncMap{
+			"extendDict": func(m map[string]interface{}, params ...interface{}) (map[string]interface{}, error) {
+				lenParams := len(params)
+				if lenParams%2 != 0 {
+					return nil, core.ErrRangeError
+				}
+
+				for i := 0; i < lenParams; i += 2 {
+					key, ok := params[i].(string)
+					if !ok {
+						return nil, core.ErrTypeError
+					}
+
+					m[key] = params[i+1]
+				}
+
+				return m, nil
+			},
+		},
+	}
+
+	// Read trading data json.
+	trade, err := readTradeData("trade.json")
```

Comment:
Please move the reading of the trading data code before the `// Draw main content template.` comment.